Design.md

```diff
@@ -227,8 +229,23 @@ This API will enable or disable user on Vault.
 
 
 ### Create S3 Credential
-This API creates S3 credential for the user.
-1. `generate-access-key` api will be called using assumed role credentials.
+This API creates S3 credentials for the user.
+1. The Generate Access Key API will be called using assumed role credentials.
+1. Encrypt the secret key (For more, see [SecretKey-Encryption-Strategy](#SecretKey-Encryption-Strategy)). Store the encrypted secret key on Redis Sentinel in the hash named "OSIS", with the hash key formatted as `<Username>__<AccessKeyID>`.
+1. Log the access key ID _(Log location and tools for logging are yet to be decided)_
+
+**Important Notes:**
+* Redis Sentinel is subject to an ongoing rolling deployment.
+* If Redis Sentinel fails or crashes, the vCloud Director's OSE service must be restarted with the following command:_
+    ```shell
+    $ ose service restart
+    ```
+  * All access keys created by OSIS before OSE restart time (`ose-restart-time`) are dead keys on the Vault DB, which OSE/OSIS cannot use and must be manually deleted from the Vault. 
+  * A log location for all the dead access key IDs created by OSIS, and a script to run manually, to delete these dead keys from Vault DB, will be provided.
```

Comment:
This 'manual interaction required' could be a bit of an issue, no? All too easy not to perform this action, amassing garbage in the Vault DB. Can't our plugin delete these upon initialization or something?

Reply:
The reason for manual interaction is that Vault DB can also have the keys created other than OSIS. Once Redis crashes/restarts,  we lose the track of keys created by OSIS. This log location helps us to identify the keys created only by OSIS and will be easy to avoid deleting the keys created by other ways like CLI